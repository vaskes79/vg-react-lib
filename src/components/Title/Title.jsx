import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FONTS, MARGIN_BTN } from '../Settings';

/**
 * Title compnents for creating teatels
 * avaliable to set custom colors
 *
 * @param {string} color - set color Title
 */

/** @component */
export const H1 = styled.h1`
	margin: 0;
	margin-bottom: ${MARGIN_BTN};
	font-family: ${FONTS.base};
	font-size: ${FONTS.sizeH1};
	font-weight: ${FONTS.weight};
	line-height: 0.71;
	color: ${({ color }) => (color ? color : '#000')};
`;
/** @component */
export const H2 = H1.withComponent('h2').extend`
  font-size: 24px;
`;
/** @component */
export const H3 = H1.withComponent('h3').extend`
  font-size: 18px;
`;
/** @component */
export const H4 = H1.withComponent('h4').extend`
  font-size: 15px;
  color: green;
`;
/** @component */
export const H5 = H1.withComponent('h4').extend`
  font-size: 15px;
  color: green;
`;
/** @component */
export const H6 = H1.withComponent('h6').extend`
  font-size: 15px;
  color: green;
`;

const Title = (props) => {
	let { type } = props;
	switch (type) {
		case 'h2':
			return <H2 {...props} />;
		case 'h3':
			return <H3 {...props} />;
		case 'h4':
			return <H4 {...props} />;
		case 'h5':
			return <H5 {...props} />;
		case 'h6':
			return <H6 {...props} />;
		default:
			return <H1 {...props} />;
	}
};

Title.propTypes = {
	/** for change color value type: string */
	color: PropTypes.string
};

/** @component */
export default Title;
