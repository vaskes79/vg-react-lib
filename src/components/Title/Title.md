#### Example using the Title component

For import the component in project

```html
import { H1 } from 'vg-react-lib';
```

Default the H1

```jsx
<H1>Title H1</H1>
```
