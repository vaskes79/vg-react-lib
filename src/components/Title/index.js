import Title, { H1, H2, H3, H4, H5, H6 } from './Title';
export { H1, H2, H3, H4, H5, H6 };
/** @component */
export default Title;
