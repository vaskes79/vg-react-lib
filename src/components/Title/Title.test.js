import { H1, H2, H3, H4, H5, H6 } from './Title.jsx'

it('should render a H1', () => {
    const wrapper = shallow(
        <H1>Hello Jest!</H1>
    );
    expect(wrapper).toMatchSnapshot();
});
