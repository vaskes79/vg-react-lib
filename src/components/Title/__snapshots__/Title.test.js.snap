// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a H1 1`] = `
<h1
  className="sc-bdVaJa jQlCVT"
>
  Hello Jest!
</h1>
`;
